packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/webview_controller_delegate.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/foundation.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:webview_flutter_platform_interface/src/v4/types/load_request_params_delegate.dart';
+
+import 'types/types.dart';
+import 'webview_platform.dart';
+
+/// Interface for a platform implementation of a web view controller.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `webview_flutter` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [WebViewCookieManagerDelegate] methods.
+abstract class WebViewControllerDelegate extends PlatformInterface {
```

Comment:
This class should take a `WebViewControllerCreationParams`, so that iOS can add parameters required at creation of a `WKWebView`. Since Android and iOS don't share any creation params, it's essentially an empty class in the platform interface. Then the platform method can be updated with `WebViewPlatform.createWebViewControllerDelegate(WebViewControllerCreationParams params)`. I procrastinated updating the code examples in the doc. I'll work on that.